src/ir/properties.h

```diff
@@ -146,24 +146,41 @@ inline Expression* getSignExtValue(Expression* curr) {
   if (curr->type != Type::i32) {
     return nullptr;
   }
-  using namespace Match;
-  int32_t leftShift = 0, rightShift = 0;
-  Expression* extended = nullptr;
-  if (matches(curr,
-              binary(ShrSInt32,
-                     binary(ShlInt32, any(&extended), i32(&leftShift)),
-                     i32(&rightShift))) &&
-      leftShift == rightShift && leftShift != 0) {
-    return extended;
+  if (auto* unary = curr->dynCast<Unary>()) {
+    if (unary->op == ExtendS8Int32 || unary->op == ExtendS16Int32) {
+      return unary->value;
+    }
+  } else {
```

Comment:
instead of an else here, we can put a `return nullptr` (like on line 147, to do an early return), and then we do not need to indent the code below.